### ISSUE TITLE
Add documentation how to upgrade

### ISSUE THREAD

**Issue by okainov** [coppermine-gallery/cpg1.6.x#25]
Readme says about "docs", but it's missing... No idea what is the process if I want to upgrade from 1.5.x

**Comment by kriegaex** [comment]
Yes, or at least add a link to existing documentation to the read-me, if there you require users to (quote) "RTFM". Back on the original web site the latest manual is for 1.5.44, so how do I upgrade from 1.5.44 to 1.6.x?

**Comment by kriegaex** [comment]
Somehow - I do not know how I ended up at that page - I found a current 1.6.x manual online:
http://coppermine-gallery.net/docs/curr/en/index.htm

There again I found this:
http://coppermine-gallery.net/docs/curr/en/upgrading.htm

But when searching for documentation links on http://coppermine-gallery.net/, als that is listed are 1.5.x and 1.4.x links, the menu there definitely also needs an update because when just clicking on "Documentation", again I end up at the 1.5.x manual. The whole web site still lists 1.5.44 as the current version and offers it for download. It is so outdated.

**Comment by eenemeenemuu** [comment]
Actually the website lists cpg1.5.46 as the current version. cpg1.5.x is still the official release due to the reason that e.g. the documentation hasn't been upgraded yet - while cpg1.6.x is ready for production (as far as we know). As soon as cpg1.6.x is the official release this issue will be obsolete.

**Comment by kriegaex** [comment]
Yes, I wrote 1.5.44, which was a typo at the end of the minor-minor version. My bad. I can see how 1.5.46 makes a "big" difference. I was just thinking, with 1.5.x needing PHP 5.6 which is unsupported and will be deactivated even by my conservative web hoster by the end of this month (which was the reason I wanted to upgrade to 1.6.04 in the first place), it would be a really good idea not to hide the latest, production-ready release in favour of a completely outdated one.

And now without irony or sarcasm: I never heard such a lame excuse before. Thanks a lot for the great "support". Anyway, I managed to upgrade somehow, so call whatever version you like "official". Just don't expect your users to understand that kind of policy.

**Comment by Jopm17** [comment]
The Coppermine website front page is rubbish and actually discourages new users. First it does not tell you anything about 1,6 being the latest even though it's been released for years now. The News section should be updated every time there is a stable release.
The hover on the download button tells you that a 2018 release is the latest when that was three releases ago. Then when you click, it brings you into git and a choice of versions which is totally confusing for newcomers as it doesn't tally with the hover text. It should simply download the latest stable version rather than going to git.
It's a good product but it won't get more users by putting people off. It's easy to go elsewhere..

**Comment by phill104** [comment]
We are working on the front page. There are only a couple of us so it is not easy.

**Comment by kriegaex** [comment]
After a radio silence of almost 2 years, "there are only a couple of us" is your answer? Seriously, another lame excuse for just not caring about documentation? Even if this was a one person project, how long could it take to update links to point to already existing, up to date information? 5 minutes, maybe 15 minutes? 30 minutes tops, I guess. Come on, give me a break!

If this was not so sad, I would say it is hilarious.

I am so happy that I am not a Coppermine user myself and only had to help a friend migrate an old version to a newer one two years ago.

**Comment by Jopm17** [comment]
As a professional software developer, I know that even when working alone or on my "hobby" work, there are certain processes to be followed when developing and releasing software and one of those is updating all the documentation to point to the current version and checking that it is correct. This includes any user facing web sites. 
Updating the download button tool tip to give the latest release version takes only a moment yet it isn't done. Adding a News item to the website to record a new release is another quick job never done. Both should be part of the ritual processes of release.

**Comment by phill104** [comment]
Joom17,

I do agree to some extent. To be honest, since Joachim left the project we have been somewhat of a rudderless ship. Ron4mac and Andre have done sterling work in getting so much of 1.6 done. My personal task is keeping the hosting and server running and up to date. We now have a chap converting the website from a collection of html pages, all on separate sub-domains, to a portal. It takes time and we all have personal lives to deal with which has unfortunately left the project blowing in the wind somewhat. It would be sad to let it die off so please bear with us. It would be great to get more volunteers on board to fire some gusto into things.

**Comment by phill104** [comment]
Joom17,

I do agree to some extent. To be honest, since Joachim left the project we have been somewhat of a rudderless ship. Ron4mac and Andre have done sterling work in getting so much of 1.6 done. My personal task is keeping the hosting and server running and up to date. We now have a chap converting the website from a collection of html pages, all on separate sub-domains, to a portal. It takes time and we all have personal lives to deal with which has unfortunately left the project blowing in the wind somewhat. It would be sad to let it die off so please bear with us. It would be great to get more volunteers on board to fire some gusto into things.

**Comment by okainov** [comment]
@phill104  have you considered hmmm... posting some kind of news/banner on the website like "Call for Developers"? Or create some top-level issue here in Github like "Looking for contributions". There was/is Hacktoberfest and if you'd have some nice `good-first-issue` issues, it might have attracted someone. 

You know, when some random stranger comes around the website, "last release: 2018" is not really optimistic message. So we question whether project is alive or not. And then one sees some activity on Github "last week", but last release is still in 2018... That's very-very suspicious. Don't be like rapidjson.

It's fine when people are coming and leaving, there is nothing to be shy about. But give people at least some information about the status. Without this it's just handing in the middle of nowhere and users are just getting annoyed by lack of response for years. Instead, you could follow any of more open ways:
- Post "Looking for new owners as we cannot/don't want to maintain it any more"
- Post "Looking for new contributions"
- ...and so on.

**Comment by phill104** [comment]
All good points and things to consider. If we can get the current html site replaced by a portal soon we can then make some updates and calls to action. 

In many ways CPG is very up to date. Security issues are fixed very rapidly. There is however a lot of tables still used in the engine and people do not like that. To update that is a massive task and something that requires a lot of time and careful effort. It is one of the reasons why the popularity of the project has waned IMO. It is easy to volunteer to take on a small task, but one as intensive as that is quite a commitment.

We also find developers like to develop, but hate documenting things which has lead to the docs being somewhat out of date.

Finally the website. It was built at a different time all in html aside from the forums. Each page has been put on a separate sub-domain too making it more complex. So one of the team is currently working on that, creating a portal so we can begin updating everything.

So if anyone would like to help in any way, we are open to offers. It is a hobby, no advertising and no money changing hands unlike many projects. It still remains popular as a gallery package and would be a real shame if it were to go the way of the dodo.
